### PR TITLE
Estelle/docs

### LINKED ISSUE #212: Add to `cargo docs`.
Prior to building an entire `mdbook`, we should make sure our `cargo docs` are getting proper attention since we can use these docs in the `mdbook` anyway. 

I will add tasks here as I discover underexplained regions of the codebase but others should feel free to add where they feel they can or suggest tasks here.

### PR DESCRIPTION
**Pull request by 0xEstelle**
Closes #212 . Added a bunch of quick docs to all the functions, traits, structs, and enums

### CODE REVIEW

**Review by 0xJepsen — APPROVED** [review]
Amazing work! Proud of you for all your work on this! It's been amazing watching you grow as an engineer learning rust!

File: crates/cli/src/commands/backtest_data.rs [hunk 1]
```diff
@@ -7,7 +7,12 @@ use bindings::uniswap_v3_pool;
 use ethers::types::U256;
 use on_chain::monitor::HistoricalMonitor;
 
-/// Save historical data from a smart contract to a csv file.
+/// Save historical sqrt_price_x96 data from a Uniswap V3 pool contract to a csv file.
+/// # Arguments
+/// * `config` - Path to the config file. (String)
```
Comment (Autoparallel) [inline crates/cli/src/commands/backtest_data.rs:12]: `String` is not the same as `&str` by the way

**Review by Autoparallel — APPROVED** [review]
Magnificent!

**Comment by Autoparallel** [comment]
@0xEstelle @0xJepsen Anything else needed to be added here? I'm good to merge.

**Pull request merged**